core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/policyservice/HttpPolicyServiceDAO.scala

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.dsde.rawls.dataaccess.policyservice
+
+import bio.terra.policy.api.TpsApi
+import bio.terra.policy.client.ApiClient
+import bio.terra.policy.model.{
+  TpsComponent,
+  TpsObjectType,
+  TpsPaoCreateRequest,
+  TpsPolicyInput,
+  TpsPolicyInputs,
+  TpsPolicyPair
+}
+import jakarta.ws.rs.client.ClientBuilder
+import org.broadinstitute.dsde.rawls.model.{ManagedGroupRef, RawlsGroupName, RawlsRequestContext, WorkspaceRequest}
+import org.broadinstitute.dsde.rawls.util.TracingUtils
+import org.glassfish.jersey.client.ClientConfig
+import org.glassfish.jersey.jnh.connector.JavaNetHttpConnectorProvider
+
+import java.util.UUID
+import scala.concurrent.{blocking, ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
+
+class HttpPolicyServiceDAO(policyServiceUrl: String)(implicit val ec: ExecutionContext) extends PolicyServiceDAO {
+  protected def getApiClient(ctx: RawlsRequestContext): ApiClient = {
+    val client: ApiClient = new ApiClient()
+
+    // By default, the client uses the `HttpUrlConnectorProvider` which relies on a workaround for
+    // PATCH endpoints that is incompatible with Java 17. Specifying a different ConnectorProvider
+    // allows us to call PATCH endpoints in TPS.
+    val clientConfig = new ClientConfig()
+    clientConfig.connectorProvider(new JavaNetHttpConnectorProvider())
+    clientConfig.register(client.getJSON)
+    client.setHttpClient(ClientBuilder.newClient(clientConfig))
+
+    TracingUtils.enableCrossServiceTracing(client.getHttpClient, ctx)
+    client.setBasePath(policyServiceUrl)
+    client.setAccessToken(ctx.userInfo.accessToken.token)
+    client
+  }
+
+  protected def getTpsApi(ctx: RawlsRequestContext): TpsApi =
+    new TpsApi(getApiClient(ctx))
+
+  def createWorkspacePao(workspaceId: UUID,
+                         workspaceRequest: WorkspaceRequest,
+                         ctx: RawlsRequestContext
+  ): Future[Unit] = Future {
+    blocking {
+      val req =
+        new TpsPaoCreateRequest()
+          .objectType(TpsObjectType.WORKSPACE)
+          .objectId(workspaceId)
+          .component(TpsComponent.RAWLS)
+      val protectedDataPolicy = new TpsPolicyInput().namespace("terra").name("protected-data")
```

Comment:
I see "terra", "protected-data", and "group-constraint" strings scattered throughout Rawls … worth turning these into constants somewhere?